### PR TITLE
Automatic deadminning

### PR DESCRIPTION
**Pull request by ShizCalev**
Automatically deadmins admins on certain roles, configurable via config options and client preferences.

todo;

- [x] make it better.
- [x] make it EVEN better.

### CODE REVIEW

**Comment by DaxDupont** [comment]
Fuck yeah

**Comment by Timonkeyn** [comment]
You can kinda metagame it if you know the ic name of that admin because you will know when they die if they appear in the admeme list

**Comment by Arianya** [comment]
@ShizCalev Thanks for your work on this! It seems super useful in general.

Just to confirm, this doesn't block the admin from re-adminning if they get auto-deadminned, right? I don't think it does but my code reading is never the best.

Would it be possible to make "re-adminning automatically on death" a config option rather then the default? I can imagine many situations where you know your death is likely to be very temporary and don't want to be exposed to admin information, plus what Timonkeyn said in their comment.

**Comment by Cyprex** [comment]
Why not simply not readmin automatically instead of making it a config option?

**Comment by PKPenguin321** [comment]
I like this but I do think there is a valid concern to be had about metagaming potential by using adminwho. People did it when Joan would join the server whenever it was clock cult to metagame what mode it was, and it unironically worked.

**Comment by subject217** [comment]
Admin metagaming is usually more of a problem with the admin than the adminwho feature.

That being said. Auto-readminning on death is a mistake. Lots of times as an admin if you die and you know you're going to be cloned or something you want to avoid adminning because it would expose to to extra round information that you don't need to be seeing.

**Comment by ExcessiveUseOfCobblestone** [comment]
Update?

**Comment by optimumtact** [comment]
make it better please

File: code/datums/diseases/transformation.dm [hunk 1]
```diff
@@ -83,6 +83,8 @@
 		message_admins("[key_name_admin(C)] has taken control of ([key_name_admin(affected_mob)]) to replace a jobbanned player.")
 		affected_mob.ghostize(0)
 		affected_mob.key = C.key
+		if(C.client?.holder && (CONFIG_GET(flag/auto_deadmin_antagonists) || C.client.prefs?.toggles & DEADMIN_ANTAGONIST))
```
Comment (AnturK) [inline code/datums/diseases/transformation.dm]: Is there any reason these aren't in mind transfer or mob/Login ? Unless you don't want this to happen outside of roundstart, eg admin getting antag later in the round.

Comment (ShizCalev) [reply code/datums/diseases/transformation.dm]: Oversight.

**Comment by ShizCalev** [comment]
@AnturK Moved most of the handling to `Login()`, everything works fine in local testing from what I can see - how's it look to you?

**Comment by AnturK** [comment]
Code looks good, will throw a testmerge for a round when i get home.

**Comment by ShizCalev** [comment]
@AnturK it's been testmerged on sybil for 6 days.

**Pull request merged**